nextcord/guild_onboarding.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-License-Identifier: MIT
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional
+
+from .enums import OnboardingMode, OnboardingPromptType, try_enum
+from .partial_emoji import PartialEmoji
+
+if TYPE_CHECKING:
+    from .guild import Guild
+    from .types.guild import (
+        Onboarding as OnboardingPayload,
+        OnboardingPrompt as OnboardingPromptPayload,
+        OnboardingPromptOption as PromptOptionPayload,
+    )
+
+__all__ = (
+    "OnboardingPromptOption",
+    "OnboardingPrompt",
+    "Onboarding",
+)
+
+
+class OnboardingPromptOption:
+    """Represents an option in a guild's onboarding prompt.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The option's ID.
+    channel_ids: List[:class:`int`]
+        The IDs of the channels that the member will be added to when the option is selected.
+    role_ids: List[:class:`int`]
+        The IDs of the roles that the member will be assigned to when the option is selected.
+    emoji: :class:`PartialEmoji`
+        The option's emoji.
+    title: :class:`str`
+        The option's title.
+    description: Optional[:class:`str`]
+        The option's description.
+    """
+
+    __slots__ = (
+        "id",
+        "channel_ids",
+        "role_ids",
+        "emoji",
+        "title",
+        "description",
+    )
+
+    def __init__(self, *, data: PromptOptionPayload) -> None:
```

Comment:
These are presented by `Guild.edit_onboarding` that these should be user creatable but they are not very friendly for that. You also need to pass emoji as its parts when updating onboarding as shown below.
![image](https://github.com/nextcord/nextcord/assets/34910574/02dd8d6a-552f-4bcf-9fb5-644f625998c5)
During this, would be appreciated if you gave this a test yourself too.